conan/tools/meson/toolchain.py

```diff
@@ -314,3 +342,45 @@ def generate(self):
         else:
             self._write_native_file()
         VCVars(self._conanfile).generate()
+
+
+class MesonAppleMixin(object):
+    def __init__(self, conanfile):
+        if not cross_building(conanfile):
+            return
+        if self._base_compiler != 'apple-clang':
+            return
+        if not is_apple_os(conanfile.settings.get_safe("os")):
+            return
+
+        xcrun = XCRun(conanfile.settings)
```

Comment:
Better move XCRun first to modern approach. It should be less detection based and more configuration based.

Reply:
the XCRun is already imported from a new namespace `conan.tools.apple` or what do you mean?

Reply:
anyway, I think improvements of XCRun, if any, could be done in another, their own PR. this is outside of the scope of this PR IMO.

Reply:
The idea is that generators like this one do not auto-detect anything. XCRun is deducing things by executing some commands right? That should be moved to input configuration, like the new ``[conf]``. 

This is a general effort we are doing in 2.0, we are removing all the platform, architecture, compiler, env-vars detection in generators and helpers, and that should happen in the input configuration. Please try that approach better.

Reply:
I don't know right now how it should look like and how it should be implemented.
if you have some example, I can take a look, but my point it belongs to XCRun, not to the meson toolchain.
in other words, as I understand, XCRun will have some code to read `conf` file instead of executing commands.
I don't see any reason why it should be done inside this particular PR - it is just using XCRun, and for meson toolchain it's an implementation detail how exactly it is implemented under the hood.
I also don't understand how is it going to work in CI, e.g. in case of ConanCenter, e.g. who, when and how exactly should populate conf with the input configutation?
really, I am just relying on the existing implementations of `conan.tools` namespace here, if they are somehow bbad/buggy or don't meet the conan 2.0 requirements, let's create an another issue and improve in another PR.